Content.Server/Research/Systems/ResearchStealerSystem.cs

```diff
@@ -24,10 +26,20 @@ private void OnDoAfter(EntityUid uid, ResearchStealerComponent comp, ResearchSte
         if (!TryComp<TechnologyDatabaseComponent>(target, out var database))
             return;
 
-        var ev = new ResearchStolenEvent(uid, target, database.UnlockedTechnologies);
+        var ev = new ResearchStolenEvent(uid, target, new());
+        var count = _random.Next(comp.MinToSteal, comp.MaxToSteal + 1);
+        for (var i = 0; i < count; i++)
+        {
+            if (database.UnlockedTechnologies.Count == 0)
+                break;
+
+            var toRemove = _random.Pick(database.UnlockedTechnologies);
+            _research.TryRemoveTechnology(toRemove, (target, database));
+            ev.Techs.Add(toRemove);
```

Comment:
```suggestion
            if (_research.TryRemoveTechnology(toRemove, (target, database)))
               ev.Techs.Add(toRemove);
```
Should the ninja get credit for stealing even if the removal (somehow) fails?

Reply:
it shouldn't ever fail but sure